crates/ide/src/hover.rs

```diff
@@ -268,7 +268,9 @@ fn hover_type_fallback(
         }
     };
 
-    let res = render::type_info(sema, config, &expr_or_pat)?;
+    let res = render::type_info(sema, config, &expr_or_pat)
+        .or_else(|| render::struct_rest_pat(sema, config, &expr_or_pat))?;
+
```

Comment:
I feel like the type fallback is the wrong place for this. I think checking like is done for lints on lines 139-145 would be appropriate here as well, that is check if one of the decsended tokens' parent is a `RestPat` and whether that parent is a `RecordPat` and continue from there on

Reply:
You can get the typed parent like `token.syntax().parent().and_then(ast::RestPat::cast)`